src/ape/pytest/fixtures.py

```diff
@@ -68,6 +68,20 @@ def project(self) -> ProjectManager:
 
         return self.project_manager
 
+    @pytest.fixture
+    def multicall(self):
+        """
+        Activate and access the multicall module.
+        Must use a provider that supports ``debug_setCode``.
+        """
+        from ape_ethereum import multicall
+
+        self.provider.set_code(
```

Comment:
In a multi-provider / chain context, this fixture could get weird. It uses the provider connected at the time of the start up of the test. However, if you switch mid test, it won't change the chain:

For example:

```python
def test_multichain(multicall, networks):
    # Do stuff with multi-call and top-level provider.
    assert True

    with networks.parse_network_choice("evm-eosystem:network:provider") as provider:
        address = multicall.constants.MULTICALL3_ADDRESS
        assert provider.get_code(address)  # fails because code was not set on this chain.
```

I am thinking multi-call makes the most sense of the ecosystem and can be used in the active provider context continuously as such. This would also address https://github.com/ApeWorX/ape/pull/1189/files#r1068171185

Unless there is another way around this?

Reply:
Yeah, I could see that being the case

Reply:
somewhat hacky solution, but I basically just made a classmethod to deploy the contract if you needed it (via `debug_setCode`): 715c322

Reply:
Ya that works for now I suppose, but I may try it out to see what it feels like

Reply:
For sure, can't really test it w/o `debug_setCode` in foundry/hardhat